File: totosuki/ABC/A/310.py
[print(l[1]) if l[1] < (min(D) + l[2]) else print(min(D) + l[2]) for l in [list(map(int, input().split()))] for D in [list(map(int, input().split()))]]

N, P, Q = map(int, input().split())
D = list(map(int, input().split()))
D.sort()

print(P) if P < (D[0] + Q) else print(D[0] + Q)

File: OCH_Tao/ABC/A/310.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [, P, Q] = input[0].split(" ").map(x => parseInt(x, 10));
const D = input[1].split(" ").map(x => parseInt(x, 10));
console.log(Math.min(P, Q + Math.min(...D)));